packages/firestore/test/integration/api/database.test.ts

```diff
@@ -1447,117 +1473,112 @@ apiDescribe('Database', (persistence: boolean) => {
 
     it('WriteBatch.set() supports partials with merge', async () => {
       return withTestDb(persistence, async db => {
-        const ref = db
-          .collection('posts')
-          .doc()
-          .withConverter(postConverterMerge);
-        await ref.set(new Post('walnut', 'author'));
-        const batch = db.batch();
+        const ref = doc(collection(db, 'posts')).withConverter(
+          postConverterMerge
+        );
+        await setDoc(ref, new Post('walnut', 'author'));
+        const batch = writeBatch(db);
         batch.set(ref, { title: 'olive' }, { merge: true });
         await batch.commit();
-        const doc = await ref.get();
-        expect(doc.get('title')).to.equal('olive');
-        expect(doc.get('author')).to.equal('author');
+        const docSnap = await getDoc(ref);
+        expect(docSnap.get('title')).to.equal('olive');
+        expect(docSnap.get('author')).to.equal('author');
       });
     });
 
     it('WriteBatch.set() supports partials with mergeFields', async () => {
       return withTestDb(persistence, async db => {
-        const ref = db
-          .collection('posts')
-          .doc()
-          .withConverter(postConverterMerge);
-        await ref.set(new Post('walnut', 'author'));
-        const batch = db.batch();
+        const ref = doc(
+          collection(db, 'posts').withConverter(postConverterMerge)
+        );
```

Comment:
should this be 
```
        const ref = doc(collection(db, 'posts')).withConverter(postConverterMerge);
```

Reply:
It actually should not matter since the converter for a collection is used for all documents by default. But you are correct that what you have is the better translation of the original API. FIxed.